lib/internal/streams/writable.js

```diff
@@ -463,12 +464,17 @@ function _write(stream, chunk, encoding, cb) {
     if (typeof chunk === 'string') {
       if ((state[kState] & kDecodeStrings) !== 0) {
         chunk = Buffer.from(chunk, encoding);
+        length = chunk.length;
```

Comment:
since Buffer.byteLength already handles non-strings, it may be simpler to just always call it if it doesn't cause a performance regression